check.go

```diff
@@ -10,23 +10,25 @@ import (
 
 // CheckDomain queries the EPP server for the availability status of one or more domains.
 func (c *Conn) CheckDomain(domains ...string) (*DomainCheckResponse, error) {
-	err := c.encodeDomainCheck(domains, "")
+	err := c.encodeDomainCheck(domains, nil)
 	if err != nil {
 		return nil, err
 	}
 	return c.processDomainCheck(domains)
 }
 
-// CheckDomainLaunchPhase allows specifying a launch phase for the check request.
-func (c *Conn) CheckDomainLaunchPhase(launchPhase string, domains ...string) (*DomainCheckResponse, error) {
-	err := c.encodeDomainCheck(domains, launchPhase)
+// CheckDomainExtensions allows specifying extension data for the following:
+//  - "neulevel:unspec": a string of the Key=Value data for the unspec tag
+//  - "launch:phase": a string of the launch phase
+func (c *Conn) CheckDomainExtensions(domains []string, extData map[string]string) (*DomainCheckResponse, error) {
```

Comment:
From comment in #backend, I think the map type here works. I expect many more unique extensions from other parties to crop up. A struct would grow larger and cost its full size for every allocation, where the map can be smallish even with many different extensions.